### ISSUE TITLE
Tilt is not recreating containers with docker-compose

### ISSUE THREAD

**Issue by scalp42** [tilt-dev/tilt#5614]
### Expected Behavior
Assuming following `docker-compose.yml` and an "admin" service:

```yaml
  admin:
    image: admin_svc:latest
    ports:
      - 4550:4550
    env_file:
      - compose/admin.env
    volumes:
      - admin_deps:/app/deps:delegated
      - admin_build:/app/_build:delegated
    depends_on:
      - db

  db:
    image: postgres:13-alpine
    env_file:
      - compose/db.env
    volumes:
      - db:/var/lib/postgresql/data
    ports:
      - 5432:5432
```

Tiltfile:

```python
docker_build("admin_svc:latest",
               dockerfile='./Dockerfile',
               context='./',
              ssh=['default'])
docker_compose('./docker-compose.yml')
```

I'd assume Tilt to recreate the `admin` container when "admin" code is changed.

### Current Behavior
It seems that when it's build once with the `latest` tag, it won't recreate the container:

![image](https://user-images.githubusercontent.com/1475276/159393663-be461889-ddc2-4868-abe2-35f345e7c092.png)

You can see the status being `up 5 minutes` ago and the container been created an hour ago.

I checked the tag `5afa3f288921` with `docker images`:

![image](https://user-images.githubusercontent.com/1475276/159393973-cf27e16d-e43a-49d0-9e8f-c0f4cfc04356.png)

Of course, if I nuke the containers with `docker-compose rm -f`, it'll work but it doesn't seem right.

Here's the Tilt output after I update the code (you can see the `COPY . .` not being cached as expected):

![image](https://user-images.githubusercontent.com/1475276/159394174-bde5f231-f601-449f-a9ec-86a52dcccbcd.png)

And it won't recreate the container, just deploy a tag that doesn't actually match `latest` in that case:

![image](https://user-images.githubusercontent.com/1475276/159394269-c283eb6a-cf75-4c20-8b23-cd74792d9ea2.png)

TLDR: developers are updating code to realize that it's not actually the "latest" code being ran since the container doesn't get recreated.

Thanks a lot for the help!

**Comment by scalp42** [comment]
I tested again nuking every image:

```
docker images | grep admin | awk '{print $3}' | xargs -I% -P5 docker rmi -f %
```

Edit code and let Tilt run:

![image](https://user-images.githubusercontent.com/1475276/159395561-80d79054-e1bc-4d64-9607-13cdc0b3097a.png)

Running `docker images | grep admin` shows the new image that Tilt built?

![image](https://user-images.githubusercontent.com/1475276/159395609-729c768f-13c3-453c-bbc3-18de0ce15538.png)
